web/app/scripts/src/services/fieldsets-parser.es.js

```diff
@@ -78,13 +78,26 @@ module.service(moduleProperties.componentName, [
 
             form.fieldsets.forEach((fieldset, index) => {
                 model.push({
-                    fieldset: index + 1,
+                    fieldset: fieldset.id,
                     answers: {}
                 });
             });
 
             return model;
         }
+
+        this.parseData = function (data) {
```

Comment:
I think `parseModel` is also wrong, but I am not sure what it should be. This could be `parseResponse` or `parseDataToModel`.